azurerm/resource_arm_shared_image_version.go

```diff
@@ -0,0 +1,242 @@
+	return &schema.Resource{
+		Create: resourceArmSharedImageVersionCreateUpdate,
+		Read:   resourceArmSharedImageVersionRead,
+		Update: resourceArmSharedImageVersionCreateUpdate,
+		Delete: resourceArmSharedImageVersionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SharedImageVersionName,
+			},
+
+			"gallery_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SharedImageGalleryName,
+			},
+
+			"image_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SharedImageName,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"managed_image_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"regions": {
+				Type:             schema.TypeSet,
+				Required:         true,
+				Elem:             &schema.Schema{Type: schema.TypeString},
+				StateFunc:        azureRMNormalizeLocation,
+				DiffSuppressFunc: azureRMSuppressLocationDiff,
+			},
+
+			"exclude_from_latest": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmSharedImageVersionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).galleryImageVersionsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	imageVersion := d.Get("name").(string)
+	imageName := d.Get("image_name").(string)
+	galleryName := d.Get("gallery_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	managedImageId := d.Get("managed_image_id").(string)
+	excludeFromLatest := d.Get("exclude_from_latest").(bool)
+
+	regions := expandSharedImageVersionRegions(d)
+	tags := d.Get("tags").(map[string]interface{})
+
+	version := compute.GalleryImageVersion{
+		Location: utils.String(location),
+		GalleryImageVersionProperties: &compute.GalleryImageVersionProperties{
+			PublishingProfile: &compute.GalleryImageVersionPublishingProfile{
+				ExcludeFromLatest: utils.Bool(excludeFromLatest),
+				Regions:           regions,
+				Source: &compute.GalleryArtifactSource{
+					ManagedImage: &compute.ManagedArtifact{
+						ID: utils.String(managedImageId),
+					},
+				},
+			},
+		},
+		Tags: expandTags(tags),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, galleryName, imageName, imageVersion, version)
+	if err != nil {
+		return fmt.Errorf("Error creating Shared Image Version %q (Image %q / Gallery %q / Resource Group %q): %+v", imageVersion, imageName, galleryName, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
```

Comment:
🤔 appears this may not be sufficient when expanding the regions (needs confirmation)